### PR TITLE
Fix UI Schema JSON Generation,Give support to ballerina resource function ,Optimize Union Types, and Update 2026 License

### LINKED ISSUE #4362: Support for resource functions in the mi-module-gen tool
### Description

Current tool only supports public functions and remote functions. Support needs to be extended to resource functions.

### Version

_No response_

### LINKED ISSUE #4540: [Ballerina][MI] Resource Function Synapse Name Generation for MI Module Gen Tool
### Description

The mi-module-gen tool currently supports generating synapse artifacts for:
- **Public functions**: Regular Ballerina functions
- **Remote functions**: Functions marked with the `remote` qualifier

Support for **resource functions** is being added as per [GitHub Issue #4362](https://github.com/wso2/product-micro-integrator/issues/4362).

#### Current Behavior (Remote Functions)

For remote functions, the synapse template name can be directly derived from the function name:

```ballerina
remote function getUsers() returns User[]|error {
    // function implementation
}
```

**Synapse Name**: `getUsers` (uses function name directly)

This works because remote function names are simple identifiers that are:
- Unique within the client class
- Valid XML identifiers
- Descriptive and meaningful

#### Issue with Resource Functions

Resource functions have a different structure:

```ballerina
resource isolated function get users/[string userId]/drafts(
    map<string|string[]> headers = {}, 
    *GmailUsersDraftsListQueries queries
) returns ListDraftsResponse|error {
    oas:ListDraftsResponse draftList = check self.genClient->/users/[userId]/drafts(headers, queries);
    return convertOASListDraftsResponseToListDraftsResponse(draftList);
}
```

**Current Behavior**: The tool extracts only the HTTP method name (`get`) from `methodSymbol.getName().get()`, which results in:
- **Synapse Name**: `get` (not unique or descriptive)

**Problems**:
1. **Non-unique names**: Multiple resource functions with the same HTTP method (e.g., `get users/...`, `get posts/...`) would all generate the same synapse name `get`, causing conflicts
2. **Not descriptive**: The name doesn't reflect the resource path, making it difficult to identify which resource the synapse template represents
3. **Invalid XML names**: Resource paths contain special characters like `/`, `[`, `]` that are not valid in XML element names

#### Current Implementation

In `BalConnectorAnalyzer.java` (line 110):

```java
String functionName = methodSymbol.getName().get();
// ...
Component component = new Component(functionName, ...);
```

For resource functions, `methodSymbol.getName().get()` returns only the HTTP method verb (e.g., `"get"`, `"post"`, `"put"`, `"delete"`), not the full resource path.

#### Resource Function Structure in Ballerina

Resource functions in Ballerina have the following signature pattern:
```
resource [isolated] function <HTTP_METHOD> <resource_path>([parameters]) [returns <type>]
```

Where:
- `<HTTP_METHOD>`: HTTP verb (get, post, put, delete, patch, head, options)
- `<resource_path>`: Path segments with optional path parameters (e.g., `users/[string userId]/drafts`)
- Path parameters are enclosed in square brackets: `[string paramName]`

#### 1. Extract Resource Path from Function Signature

Need to extract the complete resource path from the resource function signature. This may require:
- Accessing the syntax tree node for the resource function
- Parsing the resource path segments
- Identifying path parameters

**Investigation Needed**:
- How to access the resource path from `MethodSymbol` or `FunctionSymbol`?
- Does the Ballerina Compiler API provide methods to extract resource paths?
- Alternative: Parse the function signature string representation

#### 2. Generate Synapse Name from HTTP Method + Resource Path

Combine the HTTP method with the resource path to create a unique, descriptive name:

**Example Transformations**:
```
get users/[string userId]/drafts
  → getUsersDrafts

post users/[string userId]/messages
  → postUsersMessages

get users/[string userId]/messages/[string messageId]
  → getUsersMessages
```

**Naming Strategy Options**:

**CamelCase with Path Segments**
- Convert HTTP method to lowercase
- Convert each path segment to PascalCase
- Remove path parameter brackets and types
- Concatenate: `get` + `Users` + `Drafts` = `getUsersDrafts`

#### 3. Handle Edge Cases

**Duplicate Names**:
If two resource functions generate the same synapse name:
```
get users/drafts
get users/drafts
```
**Solution**: Append a numeric suffix: `getUsersDrafts`, `getUsersDrafts2`

**Special Characters**:
- Path segments may contain hyphens: `user-profiles` → `UserProfiles`
- Path segments may be numeric: `v1/users` → `V1Users` or `getV1Users`

**Path Parameters**:
- Remove type information: `[string userId]` → use only `userId` for naming context
- Multiple parameters: `users/[string userId]/messages/[string msgId]` → `getUsersMessages`

**Empty Path Segments**:
- Root resource: `get /` → `getRoot` or `get`

#### 4. XML Name Validation

Ensure generated names are valid XML identifiers:
- Must start with a letter or underscore
- Can contain letters, digits, underscores, hyphens, periods
- Cannot contain spaces or special characters like `/`, `[`, `]`

**Sanitization Rules**:
- Replace invalid characters with underscores or remove them
- Ensure first character is valid
- Handle consecutive special characters

### Implementation Tasks

1. **Research Ballerina Compiler API**
   - Identify how to extract resource path from `MethodSymbol`/`FunctionSymbol`
   - Check if `ResourceMethodSymbol` or similar specialized symbol exists
   - Explore syntax tree access for resource functions

2. **Create Resource Path Extraction Utility**
   - Implement method to extract HTTP method and resource path
   - Handle different resource function signatures
   - Support both isolated and non-isolated resource functions

3. **Implement Synapse Name Generation**
   - Create utility method: `generateSynapseNameForResource(String httpMethod, String resourcePath)`
   - Apply naming strategy (CamelCase recommended)
   - Handle path parameter extraction and removal
   - Sanitize for XML validity

4. **Update Component Creation Logic**
   - Modify `BalConnectorAnalyzer.java` to use new naming logic for resource functions
   - Keep existing logic for remote and public functions
   - Add validation to ensure unique names within a connection

5. **Add Tests**
   - Test various resource path patterns
   - Test edge cases (special characters, duplicates, etc.)
   - Verify XML name validity
   - Ensure backward compatibility with remote functions

#### Scenario 1: Simple Resource Path
```ballerina
resource isolated function get users/drafts() returns Draft[]|error
```
**Generated Synapse Name**: `getUsersDrafts`

#### Scenario 2: Resource Path with Parameters
```ballerina
resource isolated function get users/[string userId]/drafts(...) returns Draft[]|error
```
**Generated Synapse Name**: `getUsersDrafts`
(Path parameter `[string userId]` is used for parameter extraction but removed from name)

#### Scenario 3: Multiple Path Segments
```ballerina
resource isolated function post users/[string userId]/messages/[string messageId]/replies(...) returns Reply|error
```
**Generated Synapse Name**: `postUsersMessagesReplies`

#### Scenario 4: Root Resource
```ballerina
resource isolated function get /() returns RootResource|error
```
**Generated Synapse Name**: `getRoot` or `get`

#### Scenario 5: Conflicting Names
```ballerina
resource isolated function get users/drafts(...) returns Draft[]|error
resource isolated function get users/drafts(...) returns Draft[]|error  // Different signature but same path
```
**Generated Synapse Names**: `getUsersDrafts`, `getUsersDrafts2`

### References

- [GitHub Issue #4362](https://github.com/wso2/product-micro-integrator/issues/4362) - Support for resource functions in mi-module-gen tool
- Ballerina Language Specification - Resource Functions
- Apache Synapse XML Template Naming Conventions

### Notes

- The current implementation in `BalConnectorAnalyzer.java` line 120 creates components with `functionName` directly
- The synapse template XML uses `{{name}}` from the component (see `functions_template.xml` line 19)
- Need to ensure the generated name is compatible with Synapse template naming requirements
- Consider impact on existing connectors that may already use resource functions (if any)

### Version

_No response_

### PR DESCRIPTION
**Pull request by iamvirul**
## Purpose

Resolves https://github.com/wso2/product-micro-integrator/issues/4362
Resolves https://github.com/wso2/product-micro-integrator/issues/4540

Issues related to malformed JSON generation in UI schemas for ballerina resource function, specifically when handling flattened attribute groups and nested parameters. Additionally, it addresses issues with `enable` conditions for union types and updates the copyright year to 2026.

## Goals

- Ensure valid JSON output for generated UI schemas by correctly handling commas and separators during attribute group flattening.
- Correctly identify and remove group prefixes for nested record fields to ensure cleaner UI labels and binding.
- Properly merge and maintain `enable` conditions when parameters are renamed or nested, especially for union types.
- Optimize single-variant unions by unwrapping them to avoid unnecessary dropdowns in the UI.
- Update license headers to reflect the year 2026.

## Approach

Modified `ConnectorSerializer.java` to:
- Implement logic for correctly adding JSON separators (commas) when attribute groups are flattened, avoiding syntax errors in the generated `.json` files.
- Enhance `removeGroupPrefix` method to handle nested paths (e.g., `http1Settings.proxy.host`) by identifying the group segment within the path.
- Add `mergeEnableConditions` to correctly combine parent and child visibility conditions.
- Add `updateEnableConditionsForUnionMembers` to ensure enable conditions remain valid when a union parameter is renamed.

Modified `ParamFactory.java` to:
- Automatically unwrap single-member unions (unions resolving to one non-nil type) to simplify the generated UI.
- Recursively handle nested unions and record types within unions.
- Deduplicate union member types to prevent duplicate options.

Update `MiCmd.java` and other headers to 2026.

## User stories

- As a user, I expect the generated MI connector UI schema to be valid JSON so that it can be correctly parsed and rendered by the integration tooling.
- As a user, I expect nested configuration fields (like proxy settings) to be correctly grouped and labeled in the generated UI.
- As a user, I expect conditional fields (enable/disable logic) to work correctly even for complex union types or renamed parameters.

## Release note

Fixed issues in UI schema generation related to nested attribute groups, JSON formatting during flattening, and enable condition handling for union types. Updated copyright year to 2026.

## Documentation

N/A. Internal tool logic fix.

## Training

N/A

## Certification

N/A

## Marketing

N/A

## Automation tests

- Unit tests
- Integration tests

Updated existing integration test expectations (Golden files) in `tests/src/test/resources/expected/` to match the corrected JSON output. This confirms that the changes produce the expected structure and valid JSON.

## Security checks

- Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
- Ran FindSecurityBugs plugin and verified report? yes
- Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples

N/A

## Related PRs

N/A

## Migrations (if applicable)

N/A

## Test environment

- JDK 21
- MacOS
- Ubuntu

## Learning

Analyzed the JSON generation logic in `ConnectorSerializer` and identified the edge cases where commas were missing or incorrectly added during recursive flattening of attribute groups.

### CODE REVIEW

File: tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/connectorModel/Connection.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ * Copyright (c) 2026, WSO2 LLC. (https://www.wso2.com).
```
Comment (thuva9872) [inline tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/connectorModel/Connection.java]: No need to update the year if the file is not modified.

**Comment by thuva9872** [comment]
Please squash the 44 commits in this PR into a smaller number of logical commits.

File: tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/SynapseNameGenerator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
```
Comment (thuva9872) [inline tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/SynapseNameGenerator.java]:
```suggestion
 * Copyright (c) 2026, WSO2 LLC. (https://www.wso2.com).
```

Comment (thuva9872) [reply tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/SynapseNameGenerator.java]: Check other files too.

File: tests/src/test/resources/expected/project5/functions/component.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<component name="functions" type="synapse/template">
+    <subComponents>
+        <component name="createUser">
+            <displayName>createUser</displayName>
```
Comment (thuva9872) [inline tests/src/test/resources/expected/project5/functions/component.xml]: Why is this name not sanitized like `tests/src/test/resources/expected/project4/functions/component.xml` ?

File: tests/src/test/resources/expected/project5/uischema/createUser.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
+{
```
Comment (thuva9872) [inline tests/src/test/resources/expected/project5/uischema/createUser.json:1]: is there no connection for these operations?

File: tests/src/test/resources/expected/project5/functions/component.xml (hunk 3 above)
Comment (iamvirul) [reply tests/src/test/resources/expected/project5/functions/component.xml]: When an `operationId` is present (as in project5's @openapi:ResourceInfo annotations), the displayName bypasses the `humanizeName()` call

File: tests/src/test/resources/expected/project6/functions/getItemsSearch.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<template xmlns="http://ws.apache.org/ns/synapse" name="getItemsSearch">
+    <parameter name="responseVariable" description="The target variable in which the output of the operation will be stored"/>
+    <parameter name="overwriteBody" description="Replace the Message Body in Message Context with the output payload of the operation"/>
+    <parameter name="query" description=""/>
+    <parameter name="&#x27;limit" description=""/>
```
Comment (thuva9872) [inline tests/src/test/resources/expected/project6/functions/getItemsSearch.xml]: is the encoded part needed?

File: tests/src/test/resources/expected/project5/uischema/createUser.json (hunk 4 above)
Comment (iamvirul) [reply tests/src/test/resources/expected/project5/uischema/createUser.json:1]: That's a bug need to fix

Comment (iamvirul) [reply tests/src/test/resources/expected/project5/uischema/createUser.json:1]: The `initComponent` is only set when a public init. project5's init doesn't have the public qualifier

File: tests/src/test/resources/expected/unionProject/uischema/unionProject_UnionClient.json [hunk 6]
```diff
@@ -24,15 +24,16 @@
...
+              "collapsed": true,
               "elements": [
                 {
                   "type": "attribute",
                   "value": {
                     "name": "serviceUrl",
                     "displayName": "serviceUrl",
                     "inputType": "stringOrExpression",
-                    "defaultValue": "https://test.sh",
+                    "defaultValue": "",
```
Comment (thuva9872) [inline tests/src/test/resources/expected/unionProject/uischema/unionProject_UnionClient.json]: why are the default values getting removed?

File: tests/src/test/resources/expected/project6/functions/getItemsSearch.xml (hunk 5 above)
Comment (iamvirul) [reply tests/src/test/resources/expected/project6/functions/getItemsSearch.xml]: We can skip it. In the UI schemas, I skipped that encoding

File: tool-mi-module-gen-cli/build.gradle [hunk 7]
```diff
@@ -59,6 +59,7 @@ configurations {
 dependencies {
     implementation "org.ballerinalang:ballerina-cli:${ballerinaLangVersion}"
     implementation 'info.picocli:picocli:4.0.1'
+    implementation 'com.google.code.gson:gson:2.10.1'
```
Comment (thuva9872) [inline tool-mi-module-gen-cli/build.gradle:62]: Do we need to add gson dependency separately? I think it is already coming as a transitive dependency.

File: tests/src/test/resources/expected/project6/functions/getItemsSearch.xml (hunk 5 above)
Comment (iamvirul) [reply tests/src/test/resources/expected/project6/functions/getItemsSearch.xml]: fixed

File: tests/src/test/resources/expected/project5/uischema/createUser.json (hunk 4 above)
Comment (thuva9872) [reply tests/src/test/resources/expected/project5/uischema/createUser.json:1]: the init function should be public right?

Comment (iamvirul) [reply tests/src/test/resources/expected/project5/uischema/createUser.json:1]: yes. Issue is in test project

File: tool-mi-module-gen-cli/build.gradle (hunk 7 above)
Comment (iamvirul) [reply tool-mi-module-gen-cli/build.gradle:62]: Ahh yes, Gson is already available as a transitive dependency from `ballerina-cli`. I will remove from build.gradle

File: tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/BalConnectorAnalyzer.java [hunk 8]
```diff
@@ -182,40 +203,190 @@ private void analyzeClass(Package compilePackage, ClassSymbol classSymbol, Map<S
...
 
+            // Prepare context for synapse name generation
+            SynapseNameContext.Builder contextBuilder = SynapseNameContext.builder().module(module);
+            
+            // Extract operationId from @openapi:ResourceInfo annotation if present using syntax tree API
+            Optional<String> operationIdOpt = Optional.empty();
+            try {
+                operationIdOpt = Utils.getOpenApiOperationId(methodSymbol, module, semanticModel);
+                if (operationIdOpt.isPresent()) {
+                    System.out.println("Found operationId: " + operationIdOpt.get() + " for method: " + methodSymbol.getName().orElse("<unknown>"));
```
Comment (thuva9872) [inline tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/BalConnectorAnalyzer.java]: Let's use the `printStream` that is already defined in the class level. Check other places too.

File: tool-mi-module-gen-cli/build.gradle (hunk 7 above)
Comment (iamvirul) [reply tool-mi-module-gen-cli/build.gradle:62]: but it only for runtime ? It needs to be explicitly declared as a compile-time dependency since we're directly importing its classes.

File: tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/BalConnectorAnalyzer.java (hunk 8 above)
Comment (iamvirul) [reply tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/BalConnectorAnalyzer.java]: sure

File: tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/FallbackSynapseNameGenerator.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
```
Comment (thuva9872) [inline tool-mi-module-gen-cli/src/main/java/io/ballerina/mi/analyzer/FallbackSynapseNameGenerator.java]: Check the year in all the new headers.

File: tests/src/test/resources/expected/project5/functions/component.xml (hunk 3 above)
Comment (iamvirul) [reply tests/src/test/resources/expected/project5/functions/component.xml]: fixed

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 10]
```diff
@@ -156,6 +190,196 @@ private Object createRecordValue(String jsonString, MessageContext context, int
...
+    
+    /**
+     * Finds the connection type prefix for a given record parameter name.
+     * Searches through context properties to find the pattern {CONNECTIONTYPE}_{recordParamName}_param0.
+     * 
+     * @param context The message context
+     * @param recordParamName The record parameter name (e.g., "config")
+     * @return The connection type prefix or null if not found
+     */
+    private String findConnectionTypeForParam(MessageContext context, String recordParamName) {
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java:196]: We can get the `connectionType` directly from the function stack of the message context.

File: tests/src/test/resources/expected/nestedRecordConflictProject/uischema/nestedRecordConflictProject_TestClient.json [hunk 11]
```diff
@@ -25,93 +25,101 @@
...
-                    "required": "true",
-                    "helpTip": "",
-                    "validateType": "",
-                    "matchPattern": ""
-                  }
-                }
+                      {
+                        "type": "attribute",
+                        "value": {
+                          "name": "host",
```
Comment (thuva9872) [inline tests/src/test/resources/expected/nestedRecordConflictProject/uischema/nestedRecordConflictProject_TestClient.json]: The name of the attributes should be same as the parameter name in the xml.

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 12]
```diff
@@ -147,6 +147,40 @@ private Object getUnionParameter(String paramName, MessageContext context, int i
     }
 
     private Object createRecordValue(String jsonString, MessageContext context, int paramIndex) {
+        // Check if this is a flattened record from init function or a regular JSON record
+        // If jsonString doesn't look like JSON (doesn't start with '{' or '['), it's likely a record name
+        // that needs to be reconstructed from flattened fields
+        if (!jsonString.startsWith("{") && !jsonString.startsWith("[") && !jsonString.startsWith("'")) {
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: If it is from init function, then the value will be null. This will cause NPE.

Comment (thuva9872) [reply native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: Before that it will throw a `SynapseException` in L120. We need to handle that first and then here.

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 13]
```diff
@@ -156,6 +190,120 @@ private Object createRecordValue(String jsonString, MessageContext context, int
...
+                setNestedField(recordJson, fieldPath, fieldValue, fieldType);
+            }
+            
+            fieldIndex++;
+        }
+        
+        // Log the reconstructed record for debugging
+        String recordJsonString = recordJson.toString();
+        log.info("Reconstructed record '" + recordParamName + "' with connection type '" + connectionType + "': " + recordJsonString);
+        System.out.println("Reconstructed record '" + recordParamName + "' with connection type '" + connectionType + "': " + recordJsonString);
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: Let's remove this

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 14]
```diff
@@ -156,6 +190,120 @@ private Object createRecordValue(String jsonString, MessageContext context, int
...
+            case FLOAT:
+                current.addProperty(finalField, Double.parseDouble(valueStr));
+                break;
+            case DECIMAL:
+                current.addProperty(finalField, valueStr); // Keep as string for decimals
+                break;
+            default:
+                // String and other types
+                current.addProperty(finalField, valueStr);
+                break;
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: We can remove the `DECIMAL` case.

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 15]
```diff
@@ -156,6 +190,117 @@ private Object createRecordValue(String jsonString, MessageContext context, int
...
+     * For example, "http1Settings.proxy.host" creates nested objects and sets the value.
+     * 
+     * @param jsonObject The root JSON object
+     * @param fieldPath The dot-notation path (e.g., "http1Settings.proxy.host")
+     * @param value The value to set
+     * @param fieldType The type of the field
+     */
+    private void setNestedField(com.google.gson.JsonObject jsonObject, String fieldPath, Object value, String fieldType) {
+        String[] parts = fieldPath.split("\\.");
+        com.google.gson.JsonObject current = jsonObject;
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: Let's remove the redundant reassignment.

File: native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java [hunk 16]
```diff
@@ -146,7 +146,41 @@ private Object getUnionParameter(String paramName, MessageContext context, int i
         return null;
     }
 
-    private Object createRecordValue(String jsonString, MessageContext context, int paramIndex) {
+    private Object createRecordValue(String jsonString, String paramName, MessageContext context, int paramIndex) {
+        // Check if this is a flattened record from init function or a regular JSON record
+        // If jsonString doesn't look like JSON (doesn't start with '{' or '['), it's likely a record name
+        // that needs to be reconstructed from flattened fields
+        if (jsonString == null || (!jsonString.startsWith("{") && !jsonString.startsWith("[") && !jsonString.startsWith("'"))) {
```
Comment (thuva9872) [inline native/src/main/java/io/ballerina/stdlib/mi/BalExecutor.java]: we only need the first condition, isn't it?

File: tests/src/test/resources/expected/project4/uischema/getAllUSAStatesStatus.json [hunk 17]
```diff
@@ -32,77 +32,40 @@
...
+                {
                   "type": "attribute",
                   "value": {
-                    "name": "allowNullString",
-                    "displayName": "allowNullString",
+                    "name": "allowNull",
+                    "displayName": "allowNull",
                     "inputType": "stringOrExpression",
-                    "defaultValue": "",
+                    "defaultValue": "()",
```
Comment (thuva9872) [inline tests/src/test/resources/expected/project4/uischema/getAllUSAStatesStatus.json]: why are all the fields having `()` as the default value?

Comment (iamvirul) [reply tests/src/test/resources/expected/project4/uischema/getAllUSAStatesStatus.json]: for null values lets keep as ""

**Review by thuva9872 — APPROVED** [review]
